packages/forma-36-codemod/bin/cli.js

```diff
@@ -0,0 +1,191 @@
+      chalk.yellow(
+        'Before we continue, please stash or commit your git changes.',
+      ),
+    );
+    process.exit(1);
+  }
+}
+
+function runTransform({ files, flags, parser, transformer }) {
+  const transformerPath = path.join(transformerDirectory, `${transformer}.js`);
+
+  let args = [];
+
+  const { dry, print, explicitRequire } = flags;
+
+  if (dry) {
+    args.push('--dry');
+  }
+  if (print) {
+    args.push('--print');
+  }
+
+  if (explicitRequire === 'false') {
+    args.push('--explicit-require=false');
+  }
+
+  args.push('--verbose=2');
+
+  args.push('--ignore-pattern=**/node_modules/**');
+
+  args.push('--parser', parser);
+
+  if (parser === 'tsx') {
+    args.push('--extensions=tsx,ts,jsx,js');
+  } else {
+    args.push('--extensions=jsx,js');
+  }
+
+  args = args.concat(['--transform', transformerPath]);
+
+  if (flags.jscodeshift) {
+    args = args.concat(flags.jscodeshift);
+  }
+
+  args = args.concat(files);
+
+  console.log(`Executing command: jscodeshift ${args.join(' ')}`);
+
+  const result = execa.sync(jscodeshiftExecutable, args, {
+    stdio: 'inherit',
+    stripEof: false,
+  });
+
+  if (result.error) {
+    throw result.error;
+  }
+}
+
+function expandFilePathsIfNeeded(files) {
+  const shouldExpandFiles = files.some((file) => file.includes('*'));
+  return shouldExpandFiles ? globby.sync(files) : files;
+}
+
+function run() {
+  const cli = meow(
+    {
+      description: 'Codemods for updating Forma-36 codes',
+      help: `
+    Usage
+      $ npx @contentful/forma-36-codemod <transform> <path> <...options>
+        transform        Currently only color-tokens-to-new-tokens.
+        path             Files or directory to transform. Can be a glob like src/**.test.js
+    Options
+      --force            Bypass Git repository safety checks and forcibly run codemods
+      --dry              Dry run (no changes are made to files)
+      --print            Print transformed files to your terminal
+      --jscodeshift      (Advanced) Pass options directly to jscodeshift
+    `,
+    },
+    {
+      boolean: ['force', 'dry', 'print', 'help'],
+      string: ['_'],
+      alias: {
+        h: 'help',
+      },
+    },
+  );
+
+  if (!cli.flags.dry) {
+    checkGitStatus(cli.flags.force);
+  }
+
+  if (
+    cli.input[0] &&
+    !inquirerChoices.TRANSFORMS_CHOICES.find((x) => x.value === cli.input[0])
+  ) {
+    console.error('Invalid transform choice, pick one of:');
+    console.error(
+      inquirerChoices.TRANSFORMS_CHOICES.map((x) => '- ' + x.value).join('\n'),
+    );
+    process.exit(1);
+  }
+
+  inquirer
+    .prompt([
+      {
+        type: 'input',
+        name: 'files',
+        message: 'On which files or directory should the codemods be applied?',
+        when: !cli.input[1],
+        default: '.',
+        filter: (files) => files.trim(),
+      },
+      {
+        type: 'list',
+        name: 'parser',
+        message: 'Which dialect of JavaScript do you use?',
```

Comment:
can we apply dialect based of a file extension without asking it?

Reply:
This is actually just to filter the extensions that should be transformed, if we want to, we can specify that it should run the transformer on `.ts,.tsx,.js,.jsx` by default. WDYT?

Reply:
I'd say run on `.ts,.tsx,.js,.jsx` by default

Reply:
Actually, my bad, just noticed that the parser is passed down to `recast` and is used to pretty print the file back from AST, so we need to know if it needs to use the Typescript or Javascript one.

Reply:
okay, then let's leave it